addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1971,8 +1971,21 @@ export class OdooEditor extends EventTarget {
         // Let the DOM split and delete the range.
         const contents = range.extractContents();
 
-        setSelection(start, nodeSize(start));
         const startLi = closestElement(start, 'li');
+        const endLi = closestElement(end, 'li');
+        let keepEndLi =
+            endLi &&
+            startLi &&
+            isZWS(startLi) &&
+            getListMode(startLi.parentElement) !== getListMode(endLi.parentElement);
+        if (keepEndLi) {
+            // Do not join blocks when start and end are different types of
+            // lists and the start list item only contains a zero-width space.
+            doJoin = false;
```

Comment:
Can't you test that before defining `doJoin`? So that it could stay a `const` rather than `let`.

Reply:
I have checked `isZws` for `keepEndLi`, and it will only works correctly after `extractContents`. When the start has all its contents selected, `extractContents` removes those contents, and it will replace with a `ZWS`. That’s why I am getting the value of `keepEndLi` after `extractContents`.

Reply:
Can't you check that the end LI is entirely selected? For example, I think it should be returned by `getSelectedNodes`. If it is entirely selected and the list types are different then you know you shouldn't join the lists. See the other comment below.